modules/mod_manager.ts

```diff
@@ -142,12 +151,31 @@ class ModManager {
                     
                 //Perform mod download and install.
                 try {
-                    const _url = await this.source_manager.GetFileURL();
+                    let _url = "";
+                    //TS won't let me delete this bit
+                    //This is a typeguard BTW to separate the collections from the non-collections
+                    if (!(IsInstall(this.source_manager.data))) {
```

Comment:
I'm unsure if this is the best way to allow an array of the Install type.
It may be better to change all to use an array, and for the pre existing types to get index 0. I think this constant type checking mixed in the current logic is messy.